src/backend/gl/src/device.rs

```diff
@@ -891,28 +960,81 @@ impl d::Device<B> for Device {
+                panic!(
+                    "Invalid number of color attachments: {} color_attachment of {}",
+                    color_attachment_index, max_color_attachments
+                );
+            }
+
+            let color_attachment = match color_attachment_index {
+                0 => gl::COLOR_ATTACHMENT0,
+                1 => gl::COLOR_ATTACHMENT1,
+                2 => gl::COLOR_ATTACHMENT2,
+                3 => gl::COLOR_ATTACHMENT3,
+                4 => gl::COLOR_ATTACHMENT4,
+                5 => gl::COLOR_ATTACHMENT5,
+                6 => gl::COLOR_ATTACHMENT6,
+                7 => gl::COLOR_ATTACHMENT7,
+                8 => gl::COLOR_ATTACHMENT8,
+                9 => gl::COLOR_ATTACHMENT9,
+                10 => gl::COLOR_ATTACHMENT10,
+                11 => gl::COLOR_ATTACHMENT11,
+                12 => gl::COLOR_ATTACHMENT12,
+                13 => gl::COLOR_ATTACHMENT13,
+                14 => gl::COLOR_ATTACHMENT14,
+                15 => gl::COLOR_ATTACHMENT15,
+                16 => gl::COLOR_ATTACHMENT16,
+                17 => gl::COLOR_ATTACHMENT17,
+                18 => gl::COLOR_ATTACHMENT18,
+                19 => gl::COLOR_ATTACHMENT19,
+                20 => gl::COLOR_ATTACHMENT20,
+                21 => gl::COLOR_ATTACHMENT21,
+                22 => gl::COLOR_ATTACHMENT22,
+                23 => gl::COLOR_ATTACHMENT23,
+                24 => gl::COLOR_ATTACHMENT24,
+                25 => gl::COLOR_ATTACHMENT25,
+                26 => gl::COLOR_ATTACHMENT26,
+                27 => gl::COLOR_ATTACHMENT27,
+                28 => gl::COLOR_ATTACHMENT28,
+                29 => gl::COLOR_ATTACHMENT29,
+                30 => gl::COLOR_ATTACHMENT30,
+                31 => gl::COLOR_ATTACHMENT31,
+                _ => panic!("Invalid attachment -- this shouldn't happen!"),
+            };
+
+            match attachment.format {
+                Some(Format::Rgba8Unorm) => {
+                    render_attachments.push(color_attachment);
+                    color_attachment_index += 1;
+                }
+                Some(Format::Rgba8Srgb) => {
+                    render_attachments.push(color_attachment);
+                    color_attachment_index += 1;
+                }
+                Some(Format::D32Sfloat) => render_attachments.push(gl::DEPTH_STENCIL_ATTACHMENT),
+                _ => unimplemented!(),
+            }
+        }
 
         let mut attachments_len = 0;
-        //TODO: exclude depth/stencil attachments from here
-        for (&att_point, view) in att_points.iter().zip(attachments.into_iter()) {
+        for (&render_attachment, view) in render_attachments.iter().zip(attachments.into_iter()) {
             attachments_len += 1;
             if self.share.private_caps.framebuffer_texture {
-                Self::bind_target(gl, target, att_point, view.borrow());
+                Self::bind_target(gl, target, render_attachment, view.borrow());
```

Comment:
for more complex renderpasses this won't be possible, but we can keep it as is so far